src/dft/backends/mklcpu/commit.cpp

```diff
@@ -31,113 +31,161 @@
+    DFT_ERROR status[2] = { DFTI_BAD_DESCRIPTOR };
+
+    for (auto dir : { DIR::fwd, DIR::bwd }) {
+        if (config_values.dimensions.size() == 1)
+            status[dir] = DftiCreateDescriptor(&bidirection_handle[dir], mklcpu_prec, mklcpu_dom, 1,
+                                               config_values.dimensions[0]);
+        else
+            status[dir] = DftiCreateDescriptor(&bidirection_handle[dir], mklcpu_prec, mklcpu_dom,
+                                               config_values.dimensions.size(),
+                                               config_values.dimensions.data());
     }
 
-    void commit(const detail::dft_values<prec, dom>& config_values) override {
-        set_value(handle, config_values);
-        auto status = DftiCommitDescriptor(handle);
-        if (status != DFTI_NO_ERROR) {
-            throw oneapi::mkl::exception(
-                "dft/backends/mklcpu", "commit",
-                "DftiCommitDescriptor failed with status: " + std::to_string(status));
-        }
+    if (status[0] != DFTI_NO_ERROR && status[1] != DFTI_NO_ERROR) {
+        std::string err = std::string("DftiCreateDescriptor failed with status : ") +
+                          DftiErrorMessage(status[0]) + std::string(", ") +
+                          DftiErrorMessage(status[1]);
+        throw oneapi::mkl::exception("dft/backends/mklcpu", "create_descriptor", err);
     }
+}
 
-    virtual void* get_handle() noexcept override {
-        return handle;
-    }
+template <dft::detail::precision prec, dft::detail::domain dom>
+commit_derived_impl<prec, dom>::~commit_derived_impl() {
+    for (auto dir : { DIR::fwd, DIR::bwd })
+        DftiFreeDescriptor(&bidirection_handle[dir]);
+}
 
-    virtual ~commit_derived_impl() override {
-        DftiFreeDescriptor((DFTI_DESCRIPTOR_HANDLE*)&handle);
-    }
+template <dft::detail::precision prec, dft::detail::domain dom>
+void commit_derived_impl<prec, dom>::commit(
+    const dft::detail::dft_values<prec, dom>& config_values) {
+    set_value(bidirection_handle.data(), config_values);
+
+    this->get_queue()
+        .submit([&](sycl::handler& cgh) {
+            auto bidir_handle_obj =
+                bidirection_buffer.template get_access<sycl::access::mode::read_write>(cgh);
+
+            host_task<detail::kernel_name<mklcpu_desc_t>>(cgh, [=]() {
+                MKL_LONG status[2] = { DFTI_BAD_DESCRIPTOR };
+
+                for (auto dir : { DIR::fwd, DIR::bwd })
+                    status[dir] = DftiCommitDescriptor(bidir_handle_obj[0][dir]);
+
+                if (status[0] != DFTI_NO_ERROR && status[1] != DFTI_NO_ERROR) {
+                    std::string err = std::string("DftiCommitDescriptor failed with status : ") +
+                                      DftiErrorMessage(status[0]) + std::string(", ") +
+                                      DftiErrorMessage(status[1]);
+                    throw oneapi::mkl::exception("dft/backends/mklcpu", "commit", err);
+                }
```

Comment:
```suggestion
                MKL_LONG status[2] = { DFTI_BAD_DESCRIPTOR, DFTI_BAD_DESCRIPTOR };

                for (auto dir : { DIR::fwd, DIR::bwd }) {
                    status[dir] = DftiCommitDescriptor(bidir_handle_obj[0][dir]);
                }

                if (status[0] != DFTI_NO_ERROR && status[1] != DFTI_NO_ERROR) {
                    std::string err = std::string("DftiCommitDescriptor failed with status : ") +
                                      DftiErrorMessage(status[0]) + std::string(", ") +
                                      DftiErrorMessage(status[1]);
                    throw oneapi::mkl::exception("dft/backends/mklcpu", "commit", err);
                }
```
Same as earlier comments.

Reply:
if you are referring to the initialization of status, its fixed in 86c1ec999fca920b2212e90b4dad63e56f367b10